frontend/src/templates/PageTemplate/PageTemplate.js

```diff
@@ -0,0 +1,30 @@
+
+import React from 'react';
+import { ThemeProvider } from 'styled-components';
+import PropTypes from 'prop-types';
+import { PageWrapper } from './PageTemplate_styles';
+import GlobalStyle from '../../assets/styles/GlobalStyle';
+
+import { theme } from '../../assets/styles/theme';
+import NavBar from '../../components/NavBar/NavBar';
+
+const PageTemplate = ( { children } ) => {
+   return (
+      <ThemeProvider theme={ theme }>
```

Comment:
You could move `ThemeProvider`, global page layout (only if there's one layout used in all your views) and other providers if there will be more to the `App` component. They'd be loaded once then on application start.